app/controllers/participate/login.js

```diff
@@ -15,31 +15,31 @@ export default Ember.Controller.extend({
   authenticating: false,
   actions: {
     authenticate(attrs) {
-      return this.get('session')
-        .authenticate('authenticator:jam-jwt', attrs)
-        .then(() => {
-          var surveyController = Ember.getOwner(this).lookup('controller:participate');
-          surveyController.set('studyId', attrs.password);
-          surveyController.set('participantId', attrs.username);
-          this.transitionToRoute('participate.survey.consent');
-        })
-        .catch((e) => {
-          if (e.status === 404) {
-            this.send('toggleInvalidAuth');
+      if (!this.get('locale')) {
+        this.toggleProperty('invalidLocale');
+        this.set('authenticating', false);
+        this.get('raven').captureMessage('Locale not selected');
+      } else {
+        return this.get('session')
+          .authenticate('authenticator:jam-jwt', attrs)
+          .then(() => {
+            var surveyController = Ember.getOwner(this).lookup('controller:participate');
+            surveyController.set('studyId', attrs.password);
```

Comment:
It's probably too late to change this now, and I note this is simply rearranging existing code. Purely for future note:

It's rather weird to be passing in some of the state for a route this way- it means that there's a way in which the route might not be set up correctly if visited directly, or transitioned to sans boilerplate. 

Information you want to be accessible globally would normally be stored in either a service, or as url parameters for the route. (Since this technically involves a password, storing it via a service would be the way to go)